nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -761,28 +777,174 @@ in {
         forceSSL = lib.mkDefault true;
         enableACME = lib.mkDefault true;
 
-        locations."/system/".alias = "/var/lib/mastodon/public-system/";
-
         locations."/" = {
           tryFiles = "$uri @proxy";
+          priority = 1100;
         };
 
-        locations."@proxy" = {
-          proxyPass = (if cfg.enableUnixSocket then "http://unix:/run/mastodon-web/web.socket" else "http://127.0.0.1:${toString(cfg.webPort)}");
+        locations."= /sw.js" = {
+          tryFiles = "$uri =404";
+          priority = 2110;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=604800, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."= /sw.js.map" = {
+          tryFiles = "$uri =404";
+          priority = 2120;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=604800, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /assets/" = {
+          tryFiles = "$uri =404";
+          priority = 2230;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /avatars/" = {
+          tryFiles = "$uri =404";
+          priority = 2240;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /emoji/" = {
+          tryFiles = "$uri =404";
+          priority = 2250;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /headers/" = {
+          tryFiles = "$uri =404";
+          priority = 2260;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /packs/" = {
+          tryFiles = "$uri =404";
+          priority = 2270;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /sounds/" = {
+          tryFiles = "$uri =404";
+          priority = 2280;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, must-revalidate';
+            ${nginxCommonHeaders}
+          '';
+        };
+
+        locations."^~ /system/" = {
+          tryFiles = "$uri =404";
+          alias = "/var/lib/mastodon/public-system/";
+          priority = 2290;
+
+          extraConfig = ''
+            add_header Cache-Control 'public, max-age=2419200, immutable';
+            ${nginxCommonHeaders}
+          '';
+        };
```

Comment:
Those all seem to be duplicated except of the priority. I would say we move the common things in to a variable and merge the priority into that.

```
      let 
        commonLocation = {
          tryFiles = "$uri =404";
          extraConfig = ''
            add_header Cache-Control 'public, max-age=2419200, immutable';
            ${nginxCommonHeaders}
          '';
        };
      in
        locations."^~ /system/" = commonLocatuion / {
          alias = "/var/lib/mastodon/public-system/";
          priority = 2290;
        };
```